server/passportConfig.js

```diff
@@ -0,0 +1,45 @@
+//For all the configuration for passports so we can authenticate a user
+const User = require("./userData");
+const bcrypt = require("bcryptjs");
+const localStrategy = require("passport-local").Strategy;
+
+module.exports = function (passport) {
+  passport.use(
+    //Defining local strategy for passport. So when running it, we get username and password in a done parameter
+    //Done parameter is like we have completed this step so let go to the next step
+    new localStrategy((username, password, done) => {
+      User.findOne({ username: username }, (err, user) => {
+        if (err) throw err;
+        if (!user) return done(null, false); //null is the error and false is the user
+        //if there is a user, then compare it with bcrypt's password
+        bcrypt.compare(password, user.password, (err, result) => {
+          if (err) throw err;
+          //if true, return that user and run the done function otherwise comparision fails (password is not same)
+          if (result === true) {
```

Comment:
In JS the `if` conditional is checking for boolean values (true/false) by default, so you can actually just do -
```suggestion
          if (result) {
```